### ISSUE TITLE
v3.0.0 issues

### ISSUE THREAD

**Issue by XhmikosR** [cloudflare/CloudFlare-Wordpress#60]
Hi, guys.

So, it seems v3.0.0 is severely broken for all my sites.

1. There are numerous useless console messages coming from React
2. After entering my account email and API key I get the following and the dropdown doesn't work due to the `TypeError`

```
12:20:07.350 "Download the React DevTools for a better development experience: https://fb.me/react-devtools"compiled.js:61853:9
12:20:07.388 %c action @ 12:20:07.388 CONFIG_FETCH  color: #000000;compiled.js:75323
12:20:07.389  prev state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75337:31
12:20:07.394  action Object { type: "CONFIG_FETCH" }compiled.js:75341:28
12:20:07.396  next state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75349:31
12:20:08.116 %c action @ 12:20:08.115 CONFIG_FETCH_SUCCESS  color: #000000;compiled.js:75323
12:20:08.117  prev state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75337:31
12:20:08.118  action Object { type: "CONFIG_FETCH_SUCCESS", config: Object }compiled.js:75341:28
12:20:08.119  next state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75349:31
12:20:08.122 %c action @ 12:20:08.121 CONFIG_UPDATE_BY_KEY  color: #000000;compiled.js:75323
12:20:08.123  prev state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75337:31
12:20:08.125  action Object { type: "CONFIG_UPDATE_BY_KEY", key: "absoluteUrlBase", value: "https://domain.tld/wp-content/plugins/c…" }compiled.js:75341:28
12:20:08.125  next state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75349:31
12:20:08.128 %c action @ 12:20:08.127 INTL_FETCH_TRANSLATIONS  color: #000000;compiled.js:75323
12:20:08.129  prev state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75337:31
12:20:08.131  action Object { type: "INTL_FETCH_TRANSLATIONS" }compiled.js:75341:28
12:20:08.131  next state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75349:31
12:20:08.970 %c action @ 12:20:08.969 INTL_FETCH_TRANSLATIONS_SUCCESS  color: #000000;compiled.js:75323
12:20:08.971  prev state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75337:31
12:20:08.975  action Object { type: "INTL_FETCH_TRANSLATIONS_SUCCESS", locale: "en", translations: Object }compiled.js:75341:28
12:20:08.977  next state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75349:31
12:20:09.046 %c action @ 12:20:08.980 APPLICATION_INIT  color: #000000;compiled.js:75323
12:20:09.048  prev state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75337:31
12:20:09.050  action Object { type: "APPLICATION_INIT" }compiled.js:75341:28
12:20:09.051  next state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75349:31
12:20:42.273 %c action @ 12:20:42.272 USER_LOGIN  color: #000000;compiled.js:75323
12:20:42.274  prev state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75337:31
12:20:42.278  action Object { type: "USER_LOGIN" }compiled.js:75341:28
12:20:42.280  next state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75349:31
12:20:44.204 %c action @ 12:20:44.172 USER_LOGIN_SUCCESS  color: #000000;compiled.js:75323
12:20:44.207  prev state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75337:31
12:20:44.208  action Object { type: "USER_LOGIN_SUCCESS", email: "thanasis@giannoulis.net" }compiled.js:75341:28
12:20:44.209  next state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75349:31
12:20:44.226 %c action @ 12:20:44.211 ZONES_FETCH  color: #000000;compiled.js:75323
12:20:44.227  prev state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75337:31
12:20:44.229  action Object { type: "ZONES_FETCH" }compiled.js:75341:28
12:20:44.230  next state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75349:31
12:20:44.255 %c action @ 12:20:44.235 @@router/TRANSITION  color: #000000;compiled.js:75323
12:20:44.255  prev state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75337:31
12:20:44.258  action Object { type: "@@router/TRANSITION", payload: Object }compiled.js:75341:28
12:20:44.259  next state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75349:31
12:20:44.261 %c action @ 12:20:44.238 @@router/UPDATE_LOCATION  color: #000000;compiled.js:75323
12:20:44.261  prev state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75337:31
12:20:44.263  action Object { type: "@@router/UPDATE_LOCATION", payload: Object }compiled.js:75341:28
12:20:44.264  next state Object { activeZone: Object, app: Object, config: Object, intl: Object, user: Object, notifications: Array[0], routing: Object, zones: Object, zoneAnalytics: Object, zoneDnsRecords: Object, 4 more… }compiled.js:75349:31
12:20:44.876 Blocked loading mixed active content “http://domain.tld/”[Learn More]options-general.php
12:20:44.888 TypeError: response is undefined
[Learn More]compiled.js:84318:5
```

I'm using Firefox 49 64-bit on Windows 10. WordPress 4.6.1 with PHP 7.0.10. I haven't tried another browser but it happens for all my sites.

Note that in version 1.3.x of the plugin I had development mode and HTTPS protocol rewriting set to off.

**Comment by XhmikosR** [comment]
Note that in version 1.3.x of the plugin I had development mode and HTTPS protocol rewriting set to off.

**Comment by thellimist** [comment]
Hi @XhmikosR,

Are you using WordPress Multisite?

**Comment by XhmikosR** [comment]
Nope, just regular WP installation.

**Comment by jwineman** [comment]
Hi,

There is a known issue in the active zone selector where clicking the dropdown doesn't work but you should only see one domain in the list anyway, the WordPress blog the plugin is installed on.  Is that the case?

Have you upgraded to 3.0.2 yet?

Thanks,
John

**Comment by XhmikosR** [comment]
@jwineman: the dropdown doesn't work at all for me.

I tried v3.0.0 and v3.0.2 with the same results...

![2016-09-19_20-33-03](https://cloud.githubusercontent.com/assets/349621/18642292/cf33f052-7ea8-11e6-8cc5-fb8406f96338.png)

**Comment by jwineman** [comment]
Ah, you have mixed content warnings which is blocking the request because you're trying to load the blog over HTTPS but not all the resources on the page are HTTPS, at least one is HTTP.

**Comment by XhmikosR** [comment]
Yeah, but the thing is, there isn't any mixed request from my side and v1.3.24 works fine too with the same settings...

**Comment by jwineman** [comment]
The protocol rewrite functionality was removed in 3.x which is probably why it works in 1.3.24.  We're planning on adding it back to 3.x in the next few weeks.  My assumption is that on 3.x if you look at the "Network" tab you'll see at least one resource being loaded over HTTP.

**Comment by XhmikosR** [comment]
But I have set it to off already in 1.3.24.

**Comment by XhmikosR** [comment]
Plus I verified, I don't see any resource being loaded via HTTP with 1.3.24.

**Comment by jwineman** [comment]
In your screenshot the second to last line is:
```
Blocked loading mixed active content "[BLURRED OUT]" [Learn More]
```
I think the blurred resource is the one being loaded over http.  It looks like its options-general.php

**Comment by XhmikosR** [comment]
Like I said, there is **not** anything from my side that loads via HTTP. That is a screenshot from your plugin that is only throwing the mixed content warning and only v3.x.

**Comment by jwineman** [comment]
Can you please post the unblurred value of that error and the relative url (everything after options-general.)?

**Comment by XhmikosR** [comment]
It's just that specific website's URL.

**Comment by XhmikosR** [comment]
Which happens with any of my websites.

**Comment by jwineman** [comment]
So you can omit the domain but I need everything after `options-general.` to debug more.

**Comment by XhmikosR** [comment]
There is nothing more after the domain.

**Comment by XhmikosR** [comment]
It's just `http://foo.bar`.

**Comment by jwineman** [comment]
I need the relative request url, it cuts off the full file name after `options-general.`.
<img width="165" alt="screen shot 2016-09-19 at 5 02 42 pm" src="https://cloud.githubusercontent.com/assets/6947158/18652917/09c591f8-7e8b-11e6-939b-fdfe8600e1f1.png">

**Comment by XhmikosR** [comment]
It's just `page=cloudflare`. But the problem isn't the mixed warning. It's the `TypeError`...

**Comment by XhmikosR** [comment]
Which is

```
TypeError: res is undefined
[150]</request/<()
compiled.js:7009
[157]</Request.prototype.callback()
compiled.js:8411
[157]</Request.prototype.crossDomainError()
compiled.js:8428
[157]</Request.prototype.end/xhr.onreadystatechange()
compiled.js:8491
```

**Comment by XhmikosR** [comment]
And again, you shouldn't log so many stuff for production...

**Comment by jwineman** [comment]
Yeah I think the response is undefined because the request is being blocked by the mixed content.  If I load my test environment over https options-general.php?page=cloudflare is also loaded over https. 

- Do you use CloudFlare SSL?  If so what setting?
- Disable SSL, load the plugin, does the drop down work?

**Comment by XhmikosR** [comment]
* Full (Strict)
* I can't disable SSL sorry.

**Comment by jwineman** [comment]
Can you load the page plugin over http without disabling SSL?

**Comment by XhmikosR** [comment]
No, because I have HSTS and a redirect in my server. And this is a production site.

**Comment by XhmikosR** [comment]
I say I don't think it's the mixed content issue, because I tried with Firefox allowing the unsafe resource to load. Although, I didn't see any unsafe resource being loaded after that.

It's something v3.x does that's really weird.

**Comment by jwineman** [comment]
Hey,

Can you open a ticket with [CloudFlare support](https://support.cloudflare.com/requests/new) and reference this GitHub issue please?  It'll be quicker cause I'll able to check your settings directly etc.

You're right, something weird going on.

Thanks,
John

**Comment by XhmikosR** [comment]
I tried with one of my sites that don't use SSL, still the same TypeError which makes the dropdown to stop working. But now, the error is different

```
TypeError: response.body is null
v4ResponseOk()
compiled.js:84318
[543]</asyncFetchZones/</<()
compiled.js:77516
[150]</request/<()
compiled.js:7023
[157]</Request.prototype.callback()
compiled.js:8411
[157]</Request/<()
compiled.js:8126
[161]</Emitter.prototype.emit()
compiled.js:9036
[157]</Request.prototype.end/xhr.onreadystatechange()
compiled.js:8493
```

I'll open a ticket too.

**Comment by XhmikosR** [comment]
@jwineman: I made support request #1211708  yesterday.

**Comment by XhmikosR** [comment]
Seems to be fixed with v3.0.3, no more JavaScript error. I wonder though, should the active zone dropdown show something other than the current domain? Because when I click it nothing happens; the dropdown doesn't expand.

Finally, do you still need so much console logging by default?

**Comment by jwineman** [comment]
Thats good to hear!  We fixed a few bugs in determining the site for active zone selector, sounds like that was the root cause.  The mixed content warning went away as well?

I'm not sure if the drop down should expand since the list only has one item in it.  We're planning on removing the select box and just displaying the domain for plugins that only manage one domain (ex WordPress) in future versions.

As for the console logging, we leave it enabled because it helps us debug support issues.  Is there a reason you would rather have it disabled?

By the way, I'll respond to your PRs/minification issue at some point today as well.

Thanks,
John

**Comment by XhmikosR** [comment]
I don't get any mixed warning either.

If the dropdown is supposed to not be clickable, since there's one active zone, then I believe this issue is sorted. Feel free to close support ticket #1211708 too.

As for logging, there's no reason to have so much logging enabled for production. Ideally, you should have no logging in production; it's just bad practice IMO. You can Google around the issue if you want.

**Comment by XhmikosR** [comment]
OK, there's still something wrong with *some* of my sites.

![screen shot 2016-09-21 at 22 14 44](https://cloud.githubusercontent.com/assets/349621/18725474/159de5e2-8049-11e6-958c-0b514bc0c4f8.png)

I see no options or anything... I cleared my cache, same issue.

**Comment by thellimist** [comment]
@XhmikosR 

Do you have multiple Cloudflare accounts?

In cloudflare.com dashboard you can see the domains you’ve registered to Cloudflare.

The website domain you posted in your screenshot, is it registered to Cloudflare. If so you can verify that you logged in with the correct Cloudflare account (If you have multiple accounts).

**Comment by XhmikosR** [comment]
Yeah I have multiple accounts but for those sites I use the work account
and not my personal one.

I went back to 1.3 and everything's back to normal.

If you need more info let me know. I don't want to post domains here.

**Comment by thellimist** [comment]
@XhmikosR 

You stated only some sites had this problem. 

Referring to the sites which have issues;

What are the PHP versions?
Do they belong to a single Cloudflare account? 
Are they subdomain?

**Comment by XhmikosR** [comment]
* PHP 7.0.11
* They belong to same account
* They are not subdomains

**Comment by XhmikosR** [comment]
Also, I had set HTTPS rewriting protocol to off and SSL to Full (strict).

**Comment by thellimist** [comment]
> They belong to same account

Do you have any other sites which are in the same Cloudflare account where the plugin has the issue you mentioned?

**Comment by XhmikosR** [comment]
Yes, I tried 2 of them and both have the same issue.

**Comment by thellimist** [comment]
Sorry I meant 

Do you have any other sites which are in the same Cloudflare account where the plugin **does not** the issue you mentioned?

**Comment by XhmikosR** [comment]
Yeah, I do.

But I can't spot the difference.

3 of them have SSL set to Full strict. One works, 2 fail.

**Comment by XhmikosR** [comment]
All of them have Cache set to all.

**Comment by thellimist** [comment]
I assume you already cleared the browser cache and purged the edge cache.

I'll continue investigating. If you realise anything feel free to mention.

**Comment by XhmikosR** [comment]
Yeah, I did both.

I'm still trying to find a common link too.

**Comment by thellimist** [comment]
@XhmikosR 

Do you use Windows Server for the sites where the plugin has issues? If so is your server Windows ISS?

**Comment by XhmikosR** [comment]
Nope, standard Apache 2.4 with PHP 7 and CPanel.

**Comment by jwineman** [comment]
@XhmikosR just want to confirm here as well whether 3.0.4 fixed your issue as well?

**Comment by XhmikosR** [comment]
I can't tell if the previous is fixed because with 3.0.4 I see no Settings option for the CloudFlare plugin...

Cleared cache and everything. I can only disable the CloudFlare plugin.

**Comment by XhmikosR** [comment]
Tried this with 4 of my sites. The ones I use Full (Strict) SSL and Cache Everything don't show the menu anymore, even after clearing WP cache and CF cache. The one site that does not have SSL (set to off) and no page rule to cache everything, works though.

So, I'd say this is another regression :/

**Comment by thellimist** [comment]
Does it appears when you deactivate and reactivate the plugin?

**Comment by XhmikosR** [comment]
Nope, already tried that, same result.

**Comment by XhmikosR** [comment]
I can set you up guys an admin account in one of the affected sites. Just send me an email.

**Comment by jwineman** [comment]
https://github.com/cloudflare/CloudFlare-Wordpress/issues/89 looks like its the cause of your new issue as well.  We'll release again later today but if you remove the `return true;` like in https://github.com/cloudflare/CloudFlare-Wordpress/pull/90 you can test.

**Comment by XhmikosR** [comment]
Confirmed, works!

So, we need 3.0.5 and all of the regressions I found are sorted.

**Comment by jwineman** [comment]
Yeah the root cause of all these issues is our development environments don't match "production" WordPress sites.  I think we have a good idea of what this looks like now so we're in the process of setting up a test blog that more closely mirrors what we've seen since launching 3.x.

**Comment by jwineman** [comment]
Thanks again for all your help reporting/debugging/fixing this stuff.

**Comment by XhmikosR** [comment]
@jwineman: any ETA for the next version?

**Comment by jwineman** [comment]
~3 hrs
